internal/addrs/check.go

```diff
@@ -0,0 +1,82 @@
+package addrs
+
+import "fmt"
+
+// Check is the address of a check rule within a checkable object.
+//
+// This represents the check rule globally within a configuration, and is used
+// during graph evaluation to identify a condition result object to update with
+// the result of check rule evaluation.
+//
+// The check address is not distinct from resource traversals, and check rule
+// values are not intended to be available to the langauge, so the address is
+// not Referenceable.
+type Check struct {
+	Container Checkable
+	Type      CheckType
+	Index     int
```

Comment:
I guess another interesting characteristic of these addresses is that they are not necessary durable/comparable between runs, because the identity changes depending on the placement in the configuration.

I don't think this is a big problem, but given that most (all?) other address types are durable in that way, it might be worth an explicit note in the doc comment about that.

Reply:
Good point! I added a comment to that effect.